clang/include/clang/Driver/Options.td

```diff
@@ -2944,6 +2944,10 @@ def fno_modules_validate_textual_header_includes :
   MarshallingInfoNegativeFlag<LangOpts<"ModulesValidateTextualHeaderIncludes">>,
   HelpText<"Do not enforce -fmodules-decluse and private header restrictions for textual headers. "
            "This flag will be removed in a future Clang release.">;
+defm modules_skip_diagnostic_options : BoolFOption<"modules-skip-diagnostic-options",
```

Comment:
I did cherry-pick `-fmodules-skip-diagnostic-options`, since that's needed downstream and I wanted to keep things aligned upstream. There's no such need for header search paths flag, but if we want to test the behavior, I'm happy to add that. WDYT?

Reply:
Seems worth having a test, and a (frontend) option seems like the easiest way to do that.